### ISSUE TITLE
Type definitions about files are broken

### ISSUE THREAD

**Issue by Clansty** [gram-js/gramjs#268]
Type definitions and convertions about files, like `Api.InputFile`, `FileLike`, `TypeInputMedia` are broken

For example, `InputMediaUploadedPhoto` (TypeInputMedia) can convert to type of `FileLike`, but it shows an error:

```
TS2322: Type 'InputMediaUploadedPhoto' is not assignable to type 'FileLike | FileLike[]'.   
    Type 'InputMediaUploadedPhoto' is not assignable to type 'MessageMediaDocument'.     
    Types of property 'CONSTRUCTOR_ID' are incompatible.       
    Type '505969924' is not assignable to type '2628808919'.
```

if we add `@ts-ignore`, the program runs normally

**Comment by Clansty** [comment]
In documentation, it shows:

![image](https://user-images.githubusercontent.com/18461360/155276005-f7a98429-c133-4f21-bb5b-c48f27f75c7e.png)

**Comment by painor** [comment]
ah you're right, I forgot to add them, do you want to do it in a PR ?

**Comment by Clansty** [comment]
fixed in #269
